crates/flux-refineck/src/checker.rs

```diff
@@ -900,18 +900,32 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
         span: Span,
     ) -> Result<PolyFnSig> {
         let tcx = self.genv.tcx();
-        let predicates = self
-            .genv
-            .predicates_of(self.def_id)
-            .with_span(span)?
-            .instantiate_identity()
-            .predicates;
+        let mut def_id = Some(self.def_id);
```

Comment:
You should start with a `DefId` here

```suggestion
        let mut def_id = Some(self.def_id.to_def_id());
```